### ISSUE TITLE
Why my pool of yarn couldn't shows the same and DS queue?

### ISSUE THREAD

**Issue by yefei-ddyw** [apache/incubator-dolphinscheduler#3910]
*For better global communication, please give priority to using English description, thx! *

*Please review https://dolphinscheduler.apache.org/en-us/docs/development/issue.html when describe an issue.*

**Describe the question**
Why my pool of yarn couldn't shows the same and DS queue?

**Which version of DolphinScheduler:**
 -[1.3.2-preview]

**Which version of CDH:**
6.2.1

**Additional context**
I have some questions about the queue of ds.The document shows the queue of ds is due to MapReduce,Spark and so on.But,I find that the logs of yarn(host:19888) shows the queue is tenant code of ds,not the queue of ds.Why ?
And the document shows the tenant code is the user of Linux.So,the queue of yarn is the tenant code?
Therefore,I test the tenant and queue of ds.Here are the results.
The following is the configuration of DS and Resources Pools of yarn.
![DS队列管理](https://user-images.githubusercontent.com/68679394/95965568-a8568080-0e3c-11eb-8ae9-930c376428c7.jpg)
The image is the queue of DS.
![DS用户管理](https://user-images.githubusercontent.com/68679394/95965759-e2278700-0e3c-11eb-806e-abfcd062acaf.jpg)
The image is the users of DS.
![DS租户管理](https://user-images.githubusercontent.com/68679394/95965807-ece21c00-0e3c-11eb-8d45-6a8245145408.jpg)
The image is the tenant of DS.
![yarn所有资源池](https://user-images.githubusercontent.com/68679394/95965860-04210980-0e3d-11eb-8359-3dd38e5c3eda.jpg)
The image is all the pools of yarn.
![yarn资源池放置规则1](https://user-images.githubusercontent.com/68679394/95965991-30d52100-0e3d-11eb-9ee1-2bf6ad6323b8.jpg)
The image is the resource pool placement rules of yarn.
![yarn资源池放置规则2](https://user-images.githubusercontent.com/68679394/95966174-6843cd80-0e3d-11eb-8108-0cd035492f63.jpg)
The image is the resource pool placement rules of yarn also.
![yarn日志](https://user-images.githubusercontent.com/68679394/95966233-77c31680-0e3d-11eb-8fd5-7782cba510fe.jpg)
The image is the logs of yarn.

**Requirement or improvement**
- Please describe about your requirements or improvement suggestions.

**Comment by xingchun-chen** [comment]
who can look the problem?

**Comment by chengshiwen** [comment]
Priority of queue: queue from `other args` > user queue > tenant queue

Flink: 1.3.x has bug about queue, so that the job will be submitted to the `default` queue regardless of the existing user or tenant queue. The latest dev branch has fixed it.
Spark: No problem
MR: No problem

Can this information help you explain the current problem?

**Comment by chengshiwen** [comment]
User | Tenant | User Queue   (Submitted) | Tenant Queue   (Ignored) | Yarn Queue   (Actual)
-- | -- | -- | -- | --
1 | ddyw | root.hdfs.flink | root.hdfs.mapreduce | root.hdfs.ddyw
2 | dd011 | root.hdfs.mapreduce | root.hdfs.spark | root.hdfs.dd011
3 | ddyw1 | root.hdfs.spark | root.hdfs.flink | root.hdfs.ddyw1

I guess the reason is as follows:

The tenant ddyw submits a job with user queue `root.hdfs.flink`, not the tenant queue `root.hdfs.mapreduce`, so that `root.hdfs.flink` doesn't take effect. It may be caused by queue permissions.

So the job doesn't meet rule 1 in Dynamic Resource Pool Configuration, so the second rule takes effect by using `root..[username]` with middle pool name `hdfs`.

In summary, the job submitted by the tenant ddyw will use `root.hdfs.ddyw` queue in yarn.

**Comment by chengshiwen** [comment]
Close the question since it has been replied
